bin/internal/shared.sh

```diff
@@ -125,7 +125,7 @@ function upgrade_flutter () (
   #  * pubspec.yaml last modified after pubspec.lock
   if [[ ! -f "$SNAPSHOT_PATH" || \
         ! -s "$STAMP_PATH" || \
-        "$(cat "$STAMP_PATH")" != "$compilekey" || \
+        "$(< "$STAMP_PATH")" != "$compilekey" || \
```

Comment:
While you're probably right, that your version saves some memory and cpu cycles, I don't think it's worth the hit in readability, and this file already suffers greatly from being difficult to understand. Thank you for your contribution, however, I think I am going to close this as a won't fix.

I would be happy to see any other proprosals you may have for improving the performance of the tool filed as issues, however. It would help to get the proposal approved if you provide a benchmark demonstrating the impact. Thanks!